### ISSUE TITLE
Error when using an audonnx object as processing function

### ISSUE THREAD

**Issue by hagenw** [audeering/audinterface#68]
As reported in https://github.com/audeering/w2v2-how-to/issues/13 the newest version of `audinterface` seems no longer compatible with `audonnx`.

Try the following code from https://github.com/audeering/w2v2-how-to/blob/master/notebook.ipynb

```python
import os

import audeer
import audinterface
import audonnx


model_root = 'model'
cache_root = 'cache'


audeer.mkdir(cache_root)
def cache_path(file):
    return os.path.join(cache_root, file)


url = 'https://zenodo.org/record/6221127/files/w2v2-L-robust-12.6bc4a7fd-1.1.0.zip'
dst_path = cache_path('model.zip')

if not os.path.exists(dst_path):
    audeer.download_url(
        url, 
        dst_path, 
        verbose=True,
    )
    
if not os.path.exists(model_root):
    audeer.extract_archive(
        dst_path, 
        model_root, 
        verbose=True,
    )

model = audonnx.load(model_root)

sampling_rate = 16000
interface = audinterface.Feature(
    model.outputs['logits'].labels,
    process_func=model,
    process_func_args={
        'output_names': 'logits',
    },
    sampling_rate=sampling_rate,
    resample=True,    
    verbose=True,
)
```

This results in

```
---------------------------------------------------------------------------
AttributeError                            Traceback (most recent call last)
<ipython-input-7-447aa9996e02> in <module>                                                  
----> 1 interface = audinterface.Feature(
      2     model.outputs['logits'].labels,           
      3     process_func=model,           
      4     process_func_args={
      5         'output_names': 'logits',

~/git/audeering/audinterface/audinterface/core/feature.py in new_func(*args, **kwargs)
     44                 )
     45                 warnings.warn(message, category=UserWarning, stacklevel=2)
---> 46             return func(*args, **kwargs)
     47         return new_func
     48     return _deprecated

~/git/audeering/audinterface/audinterface/core/feature.py in __init__(self, feature_names, name, params, process_func, process_func_args, process_func_is_mono, process_func_applies_sliding_wi
ndow, sampling_rate, resample, channels, mixdown, win_dur, hop_dur, min_signal_dur, max_signal_dur, segment, keep_nat, num_workers, multiprocessing, verbose, **kwargs)
    312         # add 'win_dur' and 'hop_dur' to process_func_args
    313         # if expected by function but not yet set
--> 314         signature = inspect.signature(process_func)
    315         if (
    316             'win_dur' in signature.parameters

/usr/lib/python3.8/inspect.py in signature(obj, follow_wrapped)
   3103 def signature(obj, *, follow_wrapped=True):
   3104     """Get a signature object for the passed callable."""
-> 3105     return Signature.from_callable(obj, follow_wrapped=follow_wrapped)
   3106
   3107

/usr/lib/python3.8/inspect.py in from_callable(cls, obj, follow_wrapped)
   2852     def from_callable(cls, obj, *, follow_wrapped=True):
   2853         """Constructs Signature for the given callable object."""
-> 2854         return _signature_from_callable(obj, sigcls=cls,
   2855                                         follow_wrapper_chains=follow_wrapped)
   2856

/usr/lib/python3.8/inspect.py in _signature_from_callable(obj, follow_wrapper_chains, skip_bound_arg, sigcls)
   2305                                         skip_bound_arg=skip_bound_arg)
   2306
-> 2307     if _signature_is_builtin(obj):
   2308         return _signature_from_builtin(sigcls, obj,
   2309                                        skip_bound_arg=skip_bound_arg)

/usr/lib/python3.8/inspect.py in _signature_is_builtin(obj)
   1845             # Can't test 'isinstance(type)' here, as it would
   1846             # also be True for regular python classes
-> 1847             obj in (type, object))
   1848
   1849

~/.envs/analysis/lib/python3.8/site-packages/audobject/core/object.py in __eq__(self, other)
    405
    406     def __eq__(self, other: 'Object') -> bool:
--> 407         return self.id == other.id
    408
    409     def __repr__(self) -> str:

AttributeError: type object 'type' has no attribute 'id'
```

**Comment by hagenw** [comment]
```python
obj in (type, object))
```
it seems to be that `signature` package requires that `audobject` objects cannot only compare against other objects of the same type, but need to support other objects as well (here: `type`).
So I guess we need to expand `audobject` by catching `AttributeError` or checking if it is of the right type before and return `False` then.

**Comment by frankenjoe** [comment]
Mhh, or maybe you are right and it has to do with `audobject`, cause interestingly the following works:

```python
def func(signal, sampling_rate):
    return signal.mean()


class Object:

    def __init__(
            self,
            func,
    ):
        self.func = func

    def bla(self):
        pass

    def __call__(
            self, 
            signal, 
            sampling_rate,
    ):
        return self.func(signal, sampling_rate)


object = Object(func)
audinterface.Feature(
    'mean',
    process_func=object,
)
```

**Comment by frankenjoe** [comment]
Turns out `inspect.signature(object)` in fact returns the signature of `__call__`..

```python
inspect.signature(object)
```
```
(signal, sampling_rate)
```

**Comment by frankenjoe** [comment]
Ok, you are on the right track, the above sample fails if we change to:

```python
class Object(audobject.Object):

    def __call__(
            self,
            signal,
            sampling_rate,
    ):
        return signal.mean()
```

**Comment by hagenw** [comment]
`signature` sees that it is a callable, but for some reason it then needs to test if it is a build-in function with `_signature_is_builtin(obj)` and that requires to execute `obj in (type, object))` which fails due to the current implementation of `audobject.Object.__eq__`

**Comment by frankenjoe** [comment]
Yes, and when I apply the solution you propose in https://github.com/audeering/audobject/pull/49 it indeed works:

```python
class Object(audobject.Object):

    def __call__(
            self,
            signal,
            sampling_rate,
    ):
        return signal.mean()

    def __eq__(self, other: 'Object') -> bool:
        if isinstance(other, type(self)):
            return self.id == other.id
        else:
            # it might happen that we need to compare
            # to other non-supported types, compare
            # https://github.com/audeering/audinterface/issues/68
            return False
```

**Comment by frankenjoe** [comment]
Fixed in the `audonnx>=0.6.2`.
